PresentData/PresentMonTraceConsumer.cpp

```diff
@@ -297,9 +312,15 @@ void PMTraceConsumer::HandleDxgkQueueSubmit(
                 eventIter->second->SeenDxgkPresent = true;
                 if (eventIter->second->ScreenTime != 0) {
                     CompletePresent(eventIter->second);
+                    completedPresent = true;
                 }
             }
-            mBltsByDxgContext.erase(eventIter);
+
+            if (!completedPresent)
+            {   
+                mBltsByDxgContext.erase(eventIter);

```

Comment:
I think variables like DxgKrnlHContext means more "what properties we know about this present", as opposed to PresentInDwmWaitingStruct which are about "where this present was stored". So for variables like DxgKrnlHContext that give information about the presents, it may be more intuitive and easier to debug to the number is not reset.

In the future when iterators members are added to PresentEvents, I think those will help distinguish the role of DxgKrnlHContext better.

I agree that Add() and Remove() currently feel like overkill... it's a question of personal preferences though I'd lean towards not having them yet.

For ease tracking this convo, I'm going to resolve other similar comments about resetting variables. I can apply discussion results from this thread to the other comments as needed :)